tests/integration/backend_test.py

```diff
@@ -1576,3 +1576,68 @@ def test_default_pass_serialization() -> None:
         )
         assert isinstance(reconstructed_pass, SequencePass)
         assert original_pass_dict == reconstructed_pass.to_dict()
+
+
+@pytest.mark.skipif(skip_remote_tests, reason=REASON)
+@pytest.mark.parametrize(
+    "authenticated_quum_backend_qa", [{"device_name": "H2-1E"}], indirect=True
+)
+def test_rng(authenticated_quum_backend_qa: QuantinuumBackend) -> None:
+    circ = Circuit(1, 1)
+    seed = circ.add_c_register("seed", 64)
+    bound = circ.add_c_register("bound", 32)
+    index = circ.add_c_register("index", 32)
+    num0 = circ.add_c_register("num0", 32)
+    num1 = circ.add_c_register("num1", 32)
+    num2 = circ.add_c_register("num2", 32)
+    circ.add_c_setbits([True, True, True], [seed[1], seed[3], seed[5]])  # seed = 41
+    circ.set_rng_seed(seed)
+    circ.set_rng_index(index)  # index = 0
+    circ.add_c_setbits([True, True], [bound[0], bound[2]])  # bound = 5
+    circ.set_rng_bound(bound)
+    circ.get_rng_num(num0)
+    circ.add_c_setbits([False], [bound[2]])  # bound = 1
+    circ.get_rng_num(num1)
+    circ.add_c_setbits([True], [bound[3]])  # bound = 9
+    circ.get_rng_num(num2)
+    # num0[3] should be 0, so the PhasedX should never be applied:
+    circ.PhasedX(0.5, 0.5, 0, condition_bits=[num0[3]], condition_value=1)
+    circ.Measure(Qubit(0), Bit(0))
+
+    b = authenticated_quum_backend_qa
+    h = b.process_circuit(circ, n_shots=10, language=Language.QASM)
+    r = b.get_result(h)
+    counts = r.get_counts()
+    assert len(counts) == 1  # circuit is deterministic
+    bits = next(iter(counts.keys()))
+    num0_startbit = circ.bits.index(Bit("num0", 0))
+    num1_startbit = circ.bits.index(Bit("num1", 0))
+    num2_startbit = circ.bits.index(Bit("num2", 0))
+    num0_val = sum(bits[num0_startbit + i] * pow(2, i) for i in range(32))
+    num1_val = sum(bits[num1_startbit + i] * pow(2, i) for i in range(32))
+    num2_val = sum(bits[num2_startbit + i] * pow(2, i) for i in range(32))
+    assert num0_val <= 5
+    assert num1_val <= 1
+    assert num2_val <= 9
+
+
+@pytest.mark.skipif(skip_remote_tests, reason=REASON)
+@pytest.mark.parametrize(
+    "authenticated_quum_backend_qa", [{"device_name": "H2-1E"}], indirect=True
+)
+def test_shotnum(authenticated_quum_backend_qa: QuantinuumBackend) -> None:
+    circ = Circuit(1).PhasedX(0.5, 0.5, 0).measure_all()
+    shotnum = circ.add_c_register("shotnum", 32)
+    circ.get_job_shot_num(shotnum)
+
+    b = authenticated_quum_backend_qa
+    n_shots = 10
+    h = b.process_circuit(circ, n_shots=n_shots, language=Language.QASM)
```

Comment:
Should we make the language a parameter of the test, so that we can expand this easily?

Reply:
I think we can do that when we have QIR submission working.